### ISSUE TITLE
Link between sentences mysteriously disappears.

### ISSUE THREAD

**Issue by ckjpn** [Tatoeba/tatoeba2#2764]
Link between sentences mysteriously disappears.
This may be something that has happened, or is happening elsewhere, so it might be worth looking into.
Perhaps there may be a chance that this happened with one of Sysko's early duplicate-merging scripts and isn't something that we currently need to be worried about.

See this sentence's comments and logs.

https://tatoeba.org/eng/sentences/show/21097#comment-1270458

Comments:

> Is the link between this sentence and [#194240] broken?

That used to be linked, but there is nothing in the logs that say it was unlinked.

➜ #194240
linked by an unknown member, date unknown

I have relinked this.

**Comment by jiru** [comment]
Good catch. I investigated to see if there are more of such links appearing in the logs without sentences being linked in reality. I found some interesting information.
* There are 3426 missing links on tatoeba.org and 66570 on dev.tatoeba.org. It looks like tatoeba.org got most of them removed at some point, because dev has an older copy of the database.
* On both dev and prod, the highest sentence id of such missing links is 6148643.
* On both dev and prod, I discovered that the same 1379 sentences are linked to themselves.

I compiled this list missing links into CSV files named as missing_links.* here: https://downloads.tatoeba.org/stats/

For the record, the (heavy and unoptimized!) request I used to get that data (which includes self-linked sentences) was:
```sql
with ranked_contributions as (
  select c.*, ROW_NUMBER() OVER (PARTITION BY sentence_id, translation_id ORDER BY id DESC) as rn
  from contributions as c
  inner join sentences s on s.id = c.sentence_id
  inner join sentences t on t.id = c.translation_id
  where type = 'link'
)
select CONCAT('https://tatoeba.org/sentences/show/', rc.sentence_id) as 'sentence_url', rc.sentence_id, rc.translation_id
from ranked_contributions rc
left join sentences_translations st on st.sentence_id = rc.sentence_id and st.translation_id = rc.translation_id
where rn = 1 and rc.action = 'insert' and st.id is null;
```
